### ISSUE TITLE
[a11y Design Review] IIR, Welcome Pathway

### ISSUE THREAD

**Issue by sara-amanda** [department-of-veterans-affairs/va.gov-team#93664]
# CAIA A11y Support

> [!NOTE]
> - **Review:** **FIGMA** - file to be ready week of 9/30/2024
> - **CAIA Review Date**:  10/8/2024
> - **Review Purpose**: Research Prep
> - **Reviewed By**: CAIA A11ys 

CAIA A11ys can review the Figma prototype files for desktop and mobile (if available), in order to provide feedback that may help your team successfully prepare for your upcoming research sessions. 

## Next Steps

> [!TIP]
> **DEFINITION OF DONE**: Refer to the task list in this ticket, for actionable items.

## Description

The team is currently developing a Figma prototype, that will be referenced and used when creating the coded prototype for user research. 

### Timeframe
Week of 9/30, per the initial meeting with IIR 9/25/2024. 

### No Accessibility Specialist on Team
CAIA accessibility specialists are available to provide support to IIR through this process with accessibility support, as this team does not have an accessibility specialist. 

## Details
_What details are necessary for understanding the specific work or request tracked by this issue?_

> - Creating a welcome modal for Vets when they first log in. Know that modals aren’t always the best because it obscures, but other ways they tested, folks would miss.
> - Initially wanted to direct Vets to take specific actions, but testing proved that folks were frustrated when those actions weren’t what they wanted to do. So moved to more of a welcome message instead.
> - Also want to test a prompt to update their contact information.

> - Raquel is going to review the Figma designs with the team’s designer when they return from PTO next week to make sure that’s the final design that will be turned into the coded prototype and then will send it to CAIA 

> - Majority screen readers, some zoom magnification, a voice control 

**Source**: [Initial meeting notes 9/25/2024](https://docs.google.com/document/d/1JWG2tndojw7WjJqNyvXvloEq4-psym9_JZE-REWRuAw/edit#heading=h.59hfk4bqi96g)

### Provided via CAIA Intake

- [Product outline](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/veteran-onboarding/product-outline.md)
- [User flows](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1715019181472/206d1983971576ee7696c1984666185a2fbff545?fromVisitorModal=true&sender=u4356f0478274fc3cca5b5144)
- [Prototype or mockups](https://www.figma.com/design/CertVj7cu66kFv9TnseR30/Welcome-to-My-VA?node-id=1669-19046&t=WjTSHM7O8hGmJ8I4-1) - test the following:
   - Modal Welcome
   - My VA -Prompt
   - Flow 2 - From My VA (missing two pieces of information)
   - Success Messaging
- DRAFT [AT Convo Guide (GH)](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/veteran-onboarding/assistive-tech%20research/AT%20conversation%20guide.md)
- DRAFT [AT Research Plan (GH)](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/veteran-onboarding/assistive-tech%20research/AT%20research%20plan.md)

## CAIA Assignee(s)

<details><summary>Toggle to view assignee info</summary>
- @coforma-jamie 
- @EvanAtCoforma 

</details>

## Team Info

<details><summary>Toggle to view team info</summary>

- Team name: IIR (Iterate, Innovate, Run)
- OCTO-DE product owner: Jeff Barns
- Product manager: Pete Egan
- Designer: Liza McRuer/Raquel Eisele
- FE engineer: John Rodriguez/Nick Sayre
- Product/team Slack channel: #iir-product-teams-public

</details>

## Definition of Done

```[tasklist]
### A11y Tasks
- [x] **CAIA Intake Opened** `9/13/2024`
- [x] Receive Figma Prototype from Raquel the week of 9/30/2024
- [x] Create a [Figma Prototype Review Google Doc](https://docs.google.com/document/d/1P_Rsssb52VE1AtfBMzk83vewOpopotDbFodcDX2JaDg/edit#heading=h.rzjdwvz67xuf)
- [x] Jamie's Review of the File
- [x] Evan's Review of the File
- [x] **Complete** the [Figma Prototype Review Google Doc](https://docs.google.com/document/d/1P_Rsssb52VE1AtfBMzk83vewOpopotDbFodcDX2JaDg/edit#heading=h.rzjdwvz67xuf ) and share it with the team.
- [x] **Additional Support** [Review the menu of services with the team](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/teams/CAIA/accessibility#services-we-offer).
- [ ] #92823
- [ ] #93662
- [x] [Example Findings for Comments Section](https://github.com/department-of-veterans-affairs/va.gov-team/issues/90481#issuecomment-2289272368) (Internal Ref. for CAIA a11ys) - output may differ by team and product
```

```[tasklist]
## Ticket Updates (CAIA Internal)
- [x] _Connect to an `Epic` or `Intake` (what body of work is this a part of?)_
- [x] _Label with `Originator/Team` (product team or stakeholder requesting support)_
- [x] _Label date in the `Open Date` field_
- [x] _Label with `Estimate` (level of effort expected for this ticket)_
- [x] _Add `Assignee(s)` name(s) to ticket_
- [x] _Add `Assignee(s) name(s) to each task` they will complete via handle tag (if known)_
- [x] _Select a `Priority Level`_
- [x] _Update date in `Last Checked` field_
- [ ] _Label with `Actual` (level of effort it took to complete this ticket)_
- [ ] _Update date in_ `Closed  Date`
```

**Comment by coforma-jamie** [comment]
## Findings

We have 2 issues and 2 recommendations for you, @raquelou and @emcruer:

### Issue: User doesn’t know when they’ve reached character limit

* **Screen:** [Edit Mailing](https://www.figma.com/design/CertVj7cu66kFv9TnseR30/Welcome-to-My-VA?node-id=1690-24602&node-type=frame&t=upCHjBdzIThMUiUl-0) (flow 2\)  
* **Version:** all  
* **Experience standard:** User has enough information to complete a task  
* **Description:** Street address (lines 1, 2, and 3\) have a character limit noted in the label:
![Image](https://github.com/user-attachments/assets/7596aa01-bc09-42b9-8ce5-ef675db135de)
* As far as I can tell from these screens, there’s no text letting people know how many characters they have left. This can be frustrating \- users will have to count out how many characters they’ve used, or have to stop and make corrections.   
  * Screen reader users told how many characters they have left, or that they’re out of characters. They just won’t be able to type into the field anymore.   
* **Recommendation:** One of these two options \-   
  * Remove the character limit completely if you can (I assume it’s there for a backend reason though, and won’t be possible), OR  
  * Use the [Text input with character count component](https://design.va.gov/storybook/?path=/docs/uswds-va-text-input--with-character-count) from the design system. This has a lot of great accessibility features baked in \- it announces how many characters you start with, how many are left, and when you have none left. It works equally well for sighted and non-sighted / screen reader users.

### Issue: No accessibility annotations in use

* **Screen:** all  
* **Version:** all  
* **Experience standard:** n/a  
* **Description:** These screens don’t use accessibility annotations. Therefore, there’s no way for us to check the following:  
  * [Heading levels](https://design.va.gov/about/accessibility/accessibility-annotations#heading-levels) are appropriate  
  * [Fieldsets and labels](https://design.va.gov/about/accessibility/accessibility-annotations#fieldsets--legends) are appropriately identified  
  * H1 provides enough context to determine page's purpose  
* **Recommendation:** Use VA’s [accessibility annotations](https://design.va.gov/about/accessibility/accessibility-annotations). This will make things *much* easier for the developers \- they won’t have to guess. **Let us know when you’ve added annotations, and we can review them for you\!**  
  * To be specific, annotate the following:  
    * Heading levels  
    * Fieldsets and labels  
    * Any utility styles you’re using (using a utility class to resize a heading, for example)

### Recommendation: Use updated contact info Array Data implementation

* **Screen:** Contact info ([flow 2](https://www.figma.com/design/CertVj7cu66kFv9TnseR30/Welcome-to-My-VA?node-id=1690-24515&node-type=frame&t=upCHjBdzIThMUiUl-0), [flow 4](https://www.figma.com/design/CertVj7cu66kFv9TnseR30/Welcome-to-My-VA?node-id=1673-37625&node-type=frame&t=upCHjBdzIThMUiUl-0))  
* **Version:** All  
* **Experience standard:** n/a  
* **Description:** This is just a heads up that the contact info Array Data page on the Platform site is out of date\! There’s a much improved, updated version, that includes implementation info and explains needed accessibility (headings, aria-labels, focus)  
* **Recommendation:** Use the [updated documentation](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2262) to create your contact info Array Data implementation.

### Recommendation: Use individual success messages

* **Screen:** [Contact info \- Success](https://www.figma.com/design/CertVj7cu66kFv9TnseR30/Welcome-to-My-VA?node-id=1673-37625&node-type=frame&t=upCHjBdzIThMUiUl-0)  
* **Version:** All  
* **Experience standard:** User experiences predictable interactions with components and patterns  
* **Description:** When someone makes a change to their contact information, they get a very general success message:
![Image](https://github.com/user-attachments/assets/5d005984-53e2-4c7e-9fa9-2fe46801b61d)
* This message is confusing for a few reasons:  
  * What if someone *updated* contact information, instead of *adding*?  
  * If someone has cognitive difficulties, it could be hard for them to connect “added contact info” to the more specific “updated email address.” It’s not consistent language.  
* **Recommendation:** It looks like CAIA content has [recommended](https://www.figma.com/design/CertVj7cu66kFv9TnseR30/Welcome-to-My-VA?node-id=1672-37239&node-type=frame&t=upCHjBdzIThMUiUl-0) that you use individual success messages \- the accessibility team agrees. It’s much more understandable\!
